README.md

```diff
@@ -1,21 +1,438 @@
-# Using Active Model Serializer
+# Using ActiveModel::Serializer
 
 ## Objectives
 
-  1. Understand what AMS does
-  2. Use AMS to render JSON with associated objects
-  3. Understand how AMS fits into Rails 5
-
-## Outline
-
-  * So you have all of the Post stuff working for you blog. Let's work on Author now
-  * Still in `to_json` create a `/posts/1.json` route that spits out the json
-  * Then modify it to just spit out the interesting bits
-  * Oh hey! we want to spit out the author and all of their posts. Do that with `to_json`
-  * No in the post we only want to show the interesting things
-  * So complicated. So ugly. Show the JSON result for a foursquare venue and be like "LOOK HOW TERRIBLE THIS WOULD BE WITH TO_JSON"
-  * Introduce AMS.
-  * Explain the JBuilder was a thing, but in Rails 5 a big push is to make Rails great for APIs. Changed to a more flexible AMS
-  * Explain AMS is a bunch of serializers.
-  * Show just rendering the Post, just rendering the Author
-  * Show how to embed the posts in the Author
+  1. Understand what ActiveModel::Serializer does.
+  2. Use ActiveModel::Serializer to render JSON with associated objects
+  3. Understand how ActiveModel::Serializer fits into Rails 5.
+
+## Lesson
+
+In the last iteration of our blog application, we saw that using
+`to_json` was an easy way to serialize objects to JSON without havin to
+create our own serializer. However, if we look at our current controller
+code:
+
+```ruby
+# posts_controller.rb
+# ...
+  def show
+    @post = Post.find(params[:id])
+    respond_to do |format|
+      format.html { render :show }
+      format.json { render json: @post.to_json(only: [:title, :description, :id],
+                              include: [author: { only: [:name]}]) }
+    end
+  end
+```
+
+It's clear that even a little bit of customizing the output of `to_json`
+can get ugly real quick. Imagine if the post had `comments` and comments
+had `users` and pretty soon we're getting real deep in the weeds trying
+to keep track of all the `include`s and `only`s in a single line of
+`to_json`.
```

Comment:
This paragraph does a great job of introducing students to the potential pain of too much `to_json`.